common/src/main/java/org/apache/drill/common/collections/ImmutableEntry.java

```diff
@@ -17,11 +17,11 @@
  */
 package org.apache.drill.common.collections;
 
-import java.util.Map;
-
-import com.google.common.base.Objects;
 import com.google.common.base.Preconditions;
 
+import java.util.Map;
+import java.util.Objects;
+
 public class ImmutableEntry<K, V> implements Map.Entry<K, V>  {
```

Comment:
@vvysotskyi In case you want to replace all Guava classes with JDK ones, why not first to replace Drill custom classes with the JDK equivalent?

Reply:
Good question. What about `org.apache.drill.common.collections.Collectors`? You have introduced it, when Drill has already used jdk8.

Reply:
The idea of replacing Guava classes appeared during the work on https://github.com/apache/drill/pull/1264 and changes became larger than it was expected initially.

The idea of replacing Drill custom classes is good, but it beyond the scope of the current PR, so I think it would be better to address it in a separate Jira.

Reply:
@vdiravka The purpose of `org.apache.drill.common.collections.Collectors` is to use internal iterator with a functional interface instead of creating a stream. In many cases, I'd recommend to use `Collectors` and extend this class. Using streams is justified when processing thousands of elements, for a handful number of elements, `Collectors` is a better option, IMO.

@vvysotskyi Personally, I don't see why it is necessary to replace *all* Guava classes/methods with provided by JDK. IMO, the scope of PR should be to replace only removed and deprecated by Guava team API, so it can be reduced significantly.

I don't see why it is necessary to change `ImmutableEntry` class as part of this PR when it needs to be completely replaced with the one provided by JDK.

Reply:
@vrozov sounds reasonable. Thank you for the explanation.
Does it make sense to add this info to the JavaDoc for this class and to rename this class to avoid collision with jdk `Collectors`?

Reply:
@vdiravka Yes, please feel free to add any useful information that may be missing to java doc of the `Collectors` class and/or add missing functionality. I don't think that rename is necessary (use either one).